src/tyr/isochrone_serializer.cc

```diff
@@ -142,5 +142,62 @@ std::string serializeIsochrones(const Api& request,
 
   return ss.str();
 }
+std::string serializeIsochronePbf(Api& request,
+                                  std::vector<midgard::GriddedData<2>::contour_interval_t>& intervals,
+                                  midgard::GriddedData<2>::contours_t& contours) {
+  // construct pbf output
+  Isochrone& isochrone = *request.mutable_isochrone();
+
+  // construct contours
+  for (size_t isoline_index = 0; isoline_index < contours.size(); ++isoline_index) {
+    const auto& contour = contours[isoline_index];
+    const auto& interval = intervals[isoline_index];
+
+    auto* con = isochrone.mutable_contours()->Add();
+    con->set_metric(std::get<2>(interval) == "time" ? Isochrone::time : Isochrone::distance);
+
+    con->set_metric_value(std::get<1>(interval));
+
+    // for each feature
+    for (const auto& feature : contour) {
+      // for each ring/contour
+      for (const auto& ring : feature) {
+        // construct a geometry
+        auto* geom = con->mutable_geometries()->Add();
+        for (const auto& pair : ring) {
+          geom->add_coords(round(pair.lng() * 1e6));
+          geom->add_coords(round(pair.lat() * 1e6));
+        }
+      }
+    }
+  }
+
+  return serializePbf(request);
+}
+} // namespace
+
+namespace valhalla {
+namespace tyr {
+
+std::string serializeIsochrones(Api& request,
+                                std::vector<midgard::GriddedData<2>::contour_interval_t>& intervals,
+                                midgard::GriddedData<2>::contours_t& contours,
+                                bool polygons,
+                                bool show_locations) {
+
+  // if the user requested pbf output
+  if (request.options().format() == Options_Format_pbf) {
+    return serializeIsochronePbf(request, intervals, contours);
+  }
+
+  switch (request.options().format()) {
+    case Options_Format_pbf:
+      return serializeIsochronePbf(request, intervals, contours);
+    case Options_Format_json:
+      return serializeIsochroneJson(request, intervals, contours, polygons, show_locations);
+    default:
+      throw;
```

Comment:
what is the bare `throw` doing here actually? it seems off, there's nothing catching any exception before this